xbmc/dialogs/GUIDialogFileBrowser.cpp

```diff
@@ -186,20 +186,19 @@ bool CGUIDialogFileBrowser::OnMessage(CGUIMessage& message)
         int iItem = m_viewControl.GetSelectedItem();
         int iAction = message.GetParam1();
         if (iItem < 0) break;
-        if (iAction == ACTION_SELECT_ITEM || iAction == ACTION_MOUSE_LEFT_CLICK)
+        CFileItemPtr pItem = (*m_vecItems)[iItem];
+        if ((iAction == ACTION_SELECT_ITEM || iAction == ACTION_MOUSE_LEFT_CLICK) && (!m_multipleSelection ||
+           (m_multipleSelection && (pItem->m_bIsShareOrDrive || pItem->m_bIsFolder))))
```

Comment:
ronie: NOT A OR A AND (B OR C)

phil65: NOT A OR B OR C
```
A B C |ronie phil65
-------------------
0 0 0 |  1     1
0 0 1 |  1     1
0 1 0 |  1     1
0 1 1 |  1     1
1 0 0 |  0     0
1 0 1 |  1     1
1 1 0 |  1     1
1 1 1 |  1     1
```
conclusion: /me needs to go back to school